Modules/_ctypes/_ctypes.c

```diff
@@ -5478,23 +5478,14 @@ comerror_init(PyObject *self, PyObject *args, PyObject *kwds)
 
 static int
 comerror_clear(PyObject *self) {
-    PyBaseExceptionObject *obj = (PyBaseExceptionObject *) self;
-    Py_CLEAR(obj->dict);
-    Py_CLEAR(obj->args);
-    Py_CLEAR(obj->notes);
-    Py_CLEAR(obj->traceback);
-    Py_CLEAR(obj->cause);
-    Py_CLEAR(obj->context);
-    return 0;
+    return ((PyTypeObject *)PyExc_BaseException)->tp_clear(self);
```

Comment:
Does it need cast to `PyBaseExceptionObject` (in `tp_clear` call)?
(the same question for `comerror_traverse`)

Reply:
I don't think so. In other places, people even do `tp->tp_clear((PyObject *)self);`